### PR TITLE
Add a GitHub workflow for finalizing a release

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

This automates the second half of the release process (the first being automated by `prepare-release.yml`). After this workflow completes, the only action that should remain for the releaser to do is to merge the `dev-release-*` pull request. We can't do that as part of this workflow, because CI has to finish first, and it seems pointless to create another workflow just to merge 1 PR.

Apply some of the aspects of this pipeline to `prepare-release.yml` as well:

* Make the release notes extraction process more sophisticated to work around GitHub's frustrating handling of line breaks in PR and release descriptions.

* Use GitHub app credentials in order to be able to trigger other pipelines.

### How has this been tested?

I tried the pipelines on my CVAT fork.

### Checklist

- [x] I submit my changes into the `develop` branch
- ~~[ ] I have created a changelog fragment~~ 
- ~~[ ] I have updated the documentation accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: .github/workflows/prepare-release.yml [hunk 1]
```diff
@@ -42,7 +56,7 @@ jobs:
 
       - name: Commit release preparation changes
         run: |
-          git -c user.name='github-actions[bot]' -c user.email='github-actions[bot]@users.noreply.github.com' \
+          git -c user.name='cvat-bot[bot]' -c user.email='147643061+cvat-bot[bot]@users.noreply.github.com' \
```
Comment (azhavoro) [inline .github/workflows/prepare-release.yml:59]: What that number is?

Comment (SpecLad) [reply .github/workflows/prepare-release.yml:59]: It's the user ID of the app. You can see it here.

https://api.github.com/users/cvat-bot[bot]

**Review by azhavoro — APPROVED** [review]

**Comment by azhavoro** [comment]
I had an issue yesterday with using this workflow: https://github.com/opencv/cvat/actions/runs/6618431921/job/17976838325
Do you have any ideas what's wrong?

**Comment by SpecLad** [comment]
Looks like it ran afoul of the branch protection rules. I've now added cvat-bot to the list of people who are allowed to push to `master`; that should fix it.

**Pull request merged**